### PR TITLE
Implement fe mvp

### PR DESCRIPTION
**Pull request by leowrites**
## Proposed Changes

Implement a front-end MVP for the Webstepper tool!

...

<details>
<summary>Screenshots of your changes (if applicable)</summary>

image

</details>

## Type of Change

_(Write an `X` or a brief description next to the type or types that best describe your changes.)_

| Type                                                                                    | Applies? |
| --------------------------------------------------------------------------------------- | -------- |
| 🚨 _Breaking change_ (fix or feature that would cause existing functionality to change) |          |
| ✨ _New feature_ (non-breaking change that adds functionality)                          |     x    |
| 🐛 _Bug fix_ (non-breaking change that fixes an issue)                                  |          |
| 🎨 _User interface change_ (change to user interface; provide screenshots)              |          |
| ♻️ _Refactoring_ (internal change to codebase, without changing functionality)          |          |
| 🚦 _Test update_ (change that _only_ adds or modifies tests)                            |     x    |
| 📚 _Documentation update_ (change that _only_ updates documentation)                    |          |
| 📦 _Dependency update_ (change that updates a dependency)                               |          |
| 🔧 _Internal_ (change that _only_ affects developers or continuous integration)         |          |

## Checklist

_(Complete each of the following items for your pull request. Indicate that you have completed an item by changing the `[ ]` into a `[x]` in the raw text, or by clicking on the checkbox in the rendered description on GitHub.)_

Before opening your pull request:

-   [x] I have performed a self-review of my changes.
    -   Check that all changed files included in this pull request are intentional changes.
    -   Check that all changes are relevant to the purpose of this pull request, as described above.
-   [x] I have added tests for my changes, if applicable.
    -   This is **required** for all bug fixes and new features.
-   [ ] I have updated the project documentation, if applicable.
    -   This is **required** for new features.
-   [ ] If this is my first contribution, I have added myself to the list of contributors.
-   [ ] I have updated the project Changelog (this is required for all changes).

After opening your pull request:

-   [x] I have verified that the CI tests have passed.
-   [x] I have reviewed the test coverage changes reported by Coveralls.
-   [x] I have [requested a review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) from a project maintainer.

## Questions and Comments

_(Include any questions or comments you have regarding your changes.)_

### CODE REVIEW

**Review by yoonieaj — APPROVED** [review]

File: webstepper/package.json [hunk 1]
```diff
@@ -45,10 +45,12 @@
         "@mui/icons-material": "^6.0.1",
         "@mui/material": "^6.0.1",
         "@picocss/pico": "^1.5.10",
+        "@testing-library/jest-dom": "^6.6.3",
```
Comment (david-yz-liu) [inline webstepper/package.json:48]: I assume this should be a dev dependency

File: webstepper/src/CodeDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+export default function CodeDisplay(props: CodeDisplayPropTypes) {
+    const lineBackground = (lineNumber: number) => {
+        if (lineNumber == props.highlightLine) {
+            return "#ffff00";
+        }
+    };
+
+    return (
+        <Box sx={{ width: "100%", height: "100%", backgroundColor: "white" }}>
```
Comment (david-yz-liu) [inline webstepper/src/CodeDisplay.tsx:20]: Move the styling into classes (throughout all files). This is true for both the `Box`, the individual line elements contained in it, and add a class for the highlight line.

File: webstepper/src/App.tsx [hunk 3]
```diff
@@ -1,10 +1,57 @@
-import React from "react";
+import React, { useState } from "react";
 import Header from "./Header";
+import { Button, Box, Typography, Stack } from "@mui/material";
+import SvgDisplay from "./SvgDisplay";
+import CodeDisplay from "./CodeDisplay";
 
 export default function App() {
+    const [step, setStep] = useState<number>(0);
+    const limit = 4;
```
Comment (david-yz-liu) [inline webstepper/src/App.tsx:9]: set this limit based on the number of lines in `codeText`

File: webstepper/src/App.tsx [hunk 4]
```diff
@@ -1,10 +1,57 @@
...
+    const handleStep = (newStep: number) => {
+        setStep(Math.min(Math.max(newStep, 0), limit - 1));
+    };
+    const svgPath = `/images/snapshot-${step}.svg`;
+
+    // TODO: remove this and replace it with actual stuff lol
```
Comment (david-yz-liu) [inline webstepper/src/App.tsx:15]: The TODO is fine, but the comment should be more professional

File: webstepper/src/SvgDisplay.tsx [hunk 5]
```diff
@@ -16,21 +12,23 @@ export default function SvgDisplay(props: SvgDisplayPropTypes) {
...
-    }, [props.jsonResult]);
+        const loadAndDrawSvg = async () => {
+            try {
+                const response = await fetch(props.svgPath);
+                const blob = await response.blob();
+                const image = new Image();
+                image.src = URL.createObjectURL(blob);
+                image.onload = () => {
+                    const context = canvasRef.current.getContext("2d");
+                    context.clearRect(0, 0, canvasWidth, canvasHeight);
```
Comment (david-yz-liu) [inline webstepper/src/SvgDisplay.tsx:23]: Is it not possible to use the `image` attributes `width` and `height` here?

**Review by david-yz-liu — COMMENTED** [review]
@leowrites and @yoonieaj great work, I left some inline comments for both of you to work on.

File: webstepper/src/CodeDisplay.tsx (hunk 2 above)
Comment (yoonieaj) [reply webstepper/src/CodeDisplay.tsx]: Got it! Just to confirm, should we be moving these classes to a separate style sheet?

Comment (david-yz-liu) [reply webstepper/src/CodeDisplay.tsx]: Yes, that's correct (you can introduce your own custom stylesheet for this app) 👍

**Review by david-yz-liu — APPROVED** [review]
Great job, @leowrites and @yoonieaj!

**Pull request merged**